src/main.rs

```diff
@@ -59,36 +59,27 @@ async fn main() {
     let fastboot_image = fs::read(args.image_path).expect("boot image not found");
 
     // Connect to the local SSH server
-    let tcp = match TcpStream::connect(format!("{}:{}", args.farm, args.port)) {
-        Ok(t) => t,
-        Err(e) => panic!("Server connection failed: {e}"),
-    };
-    let mut sess = Session::new().unwrap();
+    let tcp = TcpStream::connect(format!("{}:{}", args.farm, args.port))?;
```

Comment:
You might look into a crate like `anyhow` (or perhaps even abusing [`other()`](https://doc.rust-lang.org/std/io/struct.Error.html#method.other)) to inform the user _what_ you tried to connect to. Otherwise the error that you print is going to be a really generic "failed to connect". Same for other `io` operations and errors, they usually/always miss useful context.

Reply:
For `anyhow`, when `Debug`-printing, it automatically walks the `Error::source()` chain and prints them all as a stack.